### ISSUE TITLE
'CustomMSDeformableAttentionTRTP' gives incorrect outputs in Pytorch inference

### ISSUE THREAD

**Issue by Jian-danai** [DerryHub/BEVFormer_tensorrt#84]
Hi,

It looks `CustomMSDeformableAttentionTRTP `gives incorrect outputs in Pytorch inference (different from `CustomMSDeformableAttentionTRT `and `CustomMSDeformableAttention`), although the Pytorch inference result is not that important.

The reasons caused the difference are:
shapes: `reference_points` and `attention_weights` are reshaped  in TRTP, specifically, the shapes of tensors fed into the [forward funtion](https://github.com/DerryHub/BEVFormer_tensorrt/blob/6b6f3dd701c352b6fb208a391afb41896b558e19/det2trt/models/functions/multi_scale_deformable_attn.py#L30C13-L30C13)  are different
softmax: There is no softmax for `attention_weights` in `CustomMSDeformableAttentionTRTP`

Are there any reasons to give different outputs in TRTP? Or it is just a bug for Pytorch inference but does not matter? Thanks.

**Comment by DerryHub** [comment]
Softmax has been merged in OP `MultiScaleDeformableAttnTRT` of `CustomMSDeformableAttentionTRTP` for better quantization.

**Comment by Jian-danai** [comment]
Thanks. I tested the latest code it looks TRTP pytorch inference gives normal outputs, I will close this issue.
